addons/website_slides/tests/test_security.py

```diff
@@ -298,3 +299,55 @@ def test_channel_access_fields_manager(self):
         channel_superuser.invalidate_cache(['can_upload', 'can_publish'])
         self.assertTrue(channel_superuser.can_upload)
         self.assertTrue(channel_superuser.can_publish)
+
+    @mute_logger('odoo.models', 'odoo.addons.base.models.ir_rule', 'odoo.addons.base.models.ir_model')
+    def test_resource_access(self):
+        image = base64.b64encode(b'Some content')
+        channel_resources = self.env['slide.slide.resource'].with_user(self.user_officer).create([
+            {'name': 'Image ' + str(idx),
+             'slide_id': self.slide_3.id,
+             'data': image} for idx in range(2)
+        ])
+        # No public access
+        with self.assertRaises(AccessError):
+            channel_resources[0].with_user(self.user_public).read(['name'])
+        with self.assertRaises(AccessError):
+            channel_resources[0].with_user(self.user_public).write({'name': 'other name'})
+
+        # No random portal access
+        user_portal_new = self.env['res.users'].create({
```

Comment:
Any reason not to use "Patrick Portal" ?
If you don't like Patrick, I think you should let him know and work it out with him.